### ISSUE TITLE
Access denied when accessing backup logs

### ISSUE THREAD

**Issue by TechGirl-3** [vmware-tanzu/velero#6263]
### Discussed in https://github.com/vmware-tanzu/velero/discussions/6262

Hi,

I am unable to view backup logs. I get "Access denied" when I run this command "velero backup logs <<backup-name>>". I am also unable to view the backup resource list. Let me know what needs to be done here. I am using Velero-plugin-for-aws.

I really appreciate any help that can be provided.

**Comment by reasonerjt** [comment]
Hi @TechGirl-3 

Could you please clarify what version of velero you are using when you see this problem?  

Such information is fetched via a pre-signed URL:

https://github.com/vmware-tanzu/velero/blob/1fd28e8a367b4cfe8f54c74ce21a478898bc0f30/pkg/persistence/object_store.go#L561

https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/ac31cd922b3afbba651e3a82f662bbcb44128b38/velero-plugin-for-aws/object_store.go#L470

Are you using AWS S3 as your object storage?

**Comment by TechGirl-3** [comment]
Hi [reasonerjt](https://github.com/reasonerjt),

I am using Velero v1.10.0 and the object storage is S3 bucket.

Thank You.

**Comment by TechGirl-3** [comment]
Hi @reasonerjt ,

Is there any particular permission that needs to be added for Velero pod for accessing the logs for backup?

Thank You.

**Comment by reasonerjt** [comment]
@TechGirl-3 
I don't think so, as long as velero can access the bucket, it should be able to generate a pre-signed URL to download the object without credential:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html

**Comment by preston-willis** [comment]
Also having this problem on v1.12.0, any updates?

**Comment by fullykubed** [comment]
Also having this problem on v1.13.0.

**Comment by kaovilai** [comment]
For those that are reporting having this issues with aws s3, can you confirm if you're able to generate presigned url per [guide](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html)? 

> To generate a presigned URL by using the Amazon S3 console
> Sign in to the AWS Management Console and open the Amazon S3 console at https://console.aws.amazon.com/s3/.
> 
> In the left navigation pane, choose Buckets.
> 
> In the Buckets list, choose the name of the bucket that contains the object that you want a presigned URL for.
> 
> In the Objects list, select the object that you want to create a presigned URL for.
> 
> On the Object actions menu, choose Share with a presigned URL.
> 
> Specify how long you want the presigned URL to be valid.
> 
> Choose Create presigned URL.
> 
> When a confirmation appears, the URL is automatically copied to your clipboard. You will see a button to copy the presigned URL if you need to copy it again.

**Comment by fullykubed** [comment]
I have also verified that I am able to download and read the logs via the presigned urls I generate manually.

**Comment by fullykubed** [comment]
This is a pretty big usability hit for debugging issues with Velero on AWS. Is there any more information we can provide to find a root cause to this issue?

**Comment by fullykubed** [comment]
@reasonerjt
